Added detection of $TMPDIR to default params

miguelbernadi opened the pull request:
Added detection of env variable $TMPDIR to initialize contents of temp_dir when it diverges from /tmp. If the content of the variable is empty, it defaults to /tmp/ and is still overridden by flag --temp.

Signed-off-by: Miguel Bernabeu <miguel.bernabeu@bsc.es>

miguelbernadi commented:
I tested it compiled fine on my Linux workstation but it would be better to test it more. It's a proposal patch for a functionality we would like, but we can take any implementation.

Use case:
We have a cluster where our local filesystem is in a different path from /tmp/ (our /tmp is very small). When we forget to add --temp we end up filling the temporary dir and the executions fail. By default we already set $TMPDIR to the biggest partition on all our machines, so the program taking the envvar would be a net gain for us.

This change should be working in both Linux and MacOSX (and maybe other Unix derivatives) and should not change behaviour for most installations.

auton1 commented:
This looks fine. The only minor change would be that the stdlib header should be called from parameters.h.

miguelbernadi commented:
Just modified the commit to move it to parameter.h  and found out cstdlib (same header) was already included, so I have removed it to avoid needless repetition. The new patch just has the getenv call.

miguelbernadi commented:
Now it's properly commited. Previous one still had the include directive.
